pkg/vm/vm_test.go

```diff
@@ -111,6 +111,67 @@ func TestPushData4(t *testing.T) {
 
 }
 
+func TestNOTNoArgument(t *testing.T) {
+	prog := makeProgram(NOT)
+	vm := load(prog)
+	vm.Run()
+	assert.Equal(t, true, vm.state.HasFlag(faultState))
+}
+
+func TestNOTBool(t *testing.T) {
+	prog := makeProgram(NOT)
+	vm := load(prog)
+	vm.estack.PushVal(false)
+	vm.Run()
+	assert.Equal(t, false, vm.state.HasFlag(faultState))
+	assert.Equal(t, &BoolItem{true}, vm.estack.Pop().value)
+}
+
+func TestNOTNonZeroInt(t *testing.T) {
+	prog := makeProgram(NOT)
+	vm := load(prog)
+	vm.estack.PushVal(3)
+	vm.Run()
+	assert.Equal(t, false, vm.state.HasFlag(faultState))
+	assert.Equal(t, &BoolItem{false}, vm.estack.Pop().value)
+}
+
+func TestNOTArray(t *testing.T) {
+	prog := makeProgram(NOT)
+	vm := load(prog)
+	vm.estack.PushVal([]StackItem{})
+	vm.Run()
+	assert.Equal(t, false, vm.state.HasFlag(faultState))
+	assert.Equal(t, &BoolItem{false}, vm.estack.Pop().value)
+}
+
+func TestNOTStruct(t *testing.T) {
+	prog := makeProgram(NOT)
+	vm := load(prog)
+	vm.estack.Push(NewElement(&StructItem{[]StackItem{}}))
+	vm.Run()
+	assert.Equal(t, false, vm.state.HasFlag(faultState))
+	assert.Equal(t, &BoolItem{false}, vm.estack.Pop().value)
+}
+
+func TestNOTByteArray0(t *testing.T) {
```

Comment:
just curios: Is there any way we can group test in order not to duplicate code.
Having smth like:
```
tests := []struct {
		name    string
		args    args
		want    []byte
		wantErr bool
	}
t.Run(tt.name, func(t *testing.T) {..}
```
What do you think?

Reply:
Maybe. Or maybe it all is gonna be irrelevant when we're to complete #196.